data/src/client.rs

```diff
@@ -469,6 +482,35 @@ impl Client {
                     list.insert(user);
                 }
             }
+            Command::Numeric(RPL_WHOREPLY, args) => {
+                if self.supports_away_notify {
+                    // H = Here, G = gone (away)
+                    let flags = args.get(6)?.chars().collect::<Vec<char>>();
+                    let away = *(flags.first()?) == 'G';
+
+                    if let Some(list) = self.chanmap.get_mut(&args[1]) {
+                        let lookup = User::from(Nick::from(args[5].clone()));
+
+                        if let Some(mut user) = list.take(&lookup) {
+                            user.update_away(away);
+                            list.insert(user);
+                        }
+                    }
+                }
+            }
+            Command::AWAY(args) => {
+                let away = args.is_some();
+                let user = message.user()?;
+                let channels = self.user_channels(user.nickname());
+                for channel in channels.iter() {
```

Comment:
Instead of using `self.user_channels` here, I'd just `self.chanmap.values_mut()`, this should save some allocations.

Reply:
Ah great optimization.